paho/client.go

```diff
@@ -551,6 +596,73 @@ func (c *Client) close() {
 	c.debug.Println("acks tracker reset")
 }
 
+func (c *Client) readLoop() {
```

Comment:
As this is the only thing writing to `c.incommingPackets` it should close the channel when it exits (this allows anything listening on that channel to shut down cleanly). Other approaches tend to lead to things happening in an unexpected order with the listeners exiting before the sender (leading to a goroutine leak i.e. at `c.incomingPackets <- packet`).

Reply:
Agree